src/snowflake/snowpark/types.py

```diff
@@ -370,7 +370,19 @@ def _fill_ast(self, ast: proto.DataType) -> None:
 
 # Numeric types
 class _IntegralType(_NumericType):
-    pass
+    def __init__(self, **kwargs) -> None:
+        self._precision = kwargs.get("precision", None)
+
+    def __eq__(self, other):
+        def filtered(d: dict) -> dict:
+            return {k: v for k, v in d.items() if k != "_precision"}
+
+        return isinstance(other, self.__class__) and filtered(
+            self.__dict__
+        ) == filtered(other.__dict__)
```

Comment:
I think whether excluding precision or not should be based upon the context._is_snowpark_connect_mode.

scos they might want to distinguish LongType by precision